src/main/kotlin/org/ptss/support/infrastructure/repositories/ToolRepository.kt

```diff
@@ -0,0 +1,40 @@
+package org.ptss.support.infrastructure.repositories
+
+import jakarta.enterprise.context.ApplicationScoped
+import org.ptss.support.domain.interfaces.repositories.IToolRepository
+import org.ptss.support.domain.models.Tool
+import org.ptss.support.infrastructure.config.PostgreSQLConfig
+import org.ptss.support.infrastructure.persistence.entities.ToolEntity
+import java.sql.Connection
+import java.sql.DriverManager
+
+@ApplicationScoped
+class ToolRepository(
+    private val config: PostgreSQLConfig
+) : IToolRepository {
+
+    private val connection: Connection by lazy {
+        DriverManager.getConnection(config.url(), config.username(), config.password())
```

Comment:
Stop dit in een generic repository. Deze generic repository kan je dan later weer gebruiken voor de CommentRepository